core/src/main/java/hudson/model/Queue.java

```diff
@@ -1691,6 +1691,18 @@ private Runnable makeFlyWeightTaskBuildable(final BuildableItem p){
         //we double check if this is a flyweight task
         if (p.task instanceof FlyweightTask) {
             Jenkins h = Jenkins.getInstance();
+
+            Label lbl = p.getAssignedLabel();
+            String fullDisplayName = p.task.getFullDisplayName();
```

Comment:
🐛  Per https://github.com/jenkinsci/jenkins/pull/2947/ we should avoid calling getFullDisplayName unless the logger has Level.FINEST -- this method can come with some unexpected overheads due to Action evaluations.